### PR TITLE
Use `version.major` for GCC version suffix

### PR DESCRIPTION
**Pull request by SeekingMeaning**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

~~Requires Homebrew/brew#8183~~
~~Now waiting for new `brew` version > 2.4.11~~
Can be merged as of version 2.4.12

Related:
- #58520, #58522, #58524, #58525, #58526, #58527 - replace private method call
- #58558 - Fix `version_suffix`
- #58973 - Get GCC version with public method

cc @gromgit

### CODE REVIEW

**Comment by gromgit** [comment]
LGTM. Thanks much, @SeekingMeaning!

File: Formula/augustus.rb [hunk 1]
```diff
@@ -40,7 +40,7 @@ def install
     # to upstream in 2016 (see https://github.com/nextgenusfs/funannotate/issues/3).
     # See also https://github.com/Gaius-Augustus/Augustus/issues/64
     cd "src" do
-      with_env("HOMEBREW_CC" => Formula["gcc"].opt_bin/"gcc-#{Formula["gcc"].linked_version.to_s.slice(/\d+/)}") do
+      with_env("HOMEBREW_CC" => Formula["gcc"].opt_bin/"gcc-#{Formula["gcc"].installed_version.major}") do
```
Comment (SeekingMeaning) [inline Formula/augustus.rb:43]:
```
irb(main):003:0> gcc.linked_version.to_s.slice(/\d+/)
=> "10"
irb(main):004:0> gcc.installed_version.major
=> #<Version::NumericToken 10>
```

File: Formula/gromacs.rb [hunk 2]
```diff
@@ -25,8 +25,8 @@ def install
                                                              "/usr/bin/ld"
 
     args = std_cmake_args + %W[
-      -DCMAKE_C_COMPILER=gcc-#{Formula["gcc"].version_suffix}
-      -DCMAKE_CXX_COMPILER=g++-#{Formula["gcc"].version_suffix}
+      -DCMAKE_C_COMPILER=gcc-#{Formula["gcc"].installed_version.major}
+      -DCMAKE_CXX_COMPILER=g++-#{Formula["gcc"].installed_version.major}
```
Comment (SeekingMeaning) [inline Formula/gromacs.rb:29]:
```
irb(main):006:0> gcc.version_suffix
=> "10"
irb(main):007:0> gcc.installed_version.major
=> #<Version::NumericToken 10>
```

File: Formula/imake.rb [hunk 3]
```diff
@@ -52,7 +52,7 @@ def install
   test do
     # Use pipe_output because the return code is unimportant here.
     output = pipe_output("#{bin}/imake -v -s/dev/null -f/dev/null -T/dev/null 2>&1")
-    gcc_major_ver = Formula["gcc"].version_suffix
+    gcc_major_ver = Formula["gcc"].installed_version.major
```
Comment (SeekingMeaning) [inline Formula/imake.rb:55]:
```
irb(main):006:0> gcc.version_suffix
=> "10"
irb(main):007:0> gcc.installed_version.major
=> #<Version::NumericToken 10>
```

File: Formula/kahip.rb [hunk 4]
```diff
@@ -17,8 +17,8 @@ class Kahip < Formula
   depends_on "open-mpi"
 
   def install
-    ENV["CC"] = Formula["gcc"].opt_bin/"gcc-#{Formula["gcc"].version_suffix}"
-    ENV["CXX"] = Formula["gcc"].opt_bin/"g++-#{Formula["gcc"].version_suffix}"
+    ENV["CC"] = Formula["gcc"].opt_bin/"gcc-#{Formula["gcc"].installed_version.major}"
+    ENV["CXX"] = Formula["gcc"].opt_bin/"g++-#{Formula["gcc"].installed_version.major}"
```
Comment (SeekingMeaning) [inline Formula/kahip.rb:21]:
```
irb(main):006:0> gcc.version_suffix
=> "10"
irb(main):007:0> gcc.installed_version.major
=> #<Version::NumericToken 10>
```

File: Formula/lcov.rb [hunk 5]
```diff
@@ -55,8 +55,8 @@ def install
   end
 
   test do
-    gcc = Formula["gcc"].opt_bin/"gcc-#{Formula["gcc"].version_suffix}"
-    gcov = Formula["gcc"].opt_bin/"gcov-#{Formula["gcc"].version_suffix}"
+    gcc = Formula["gcc"].opt_bin/"gcc-#{Formula["gcc"].installed_version.major}"
+    gcov = Formula["gcc"].opt_bin/"gcov-#{Formula["gcc"].installed_version.major}"
```
Comment (SeekingMeaning) [inline Formula/lcov.rb:59]:
```
irb(main):006:0> gcc.version_suffix
=> "10"
irb(main):007:0> gcc.installed_version.major
=> #<Version::NumericToken 10>
```

File: Formula/pipgrip.rb [hunk 6]
```diff
@@ -53,7 +53,7 @@ def install
     venv.pip_install buildpath
 
     gcc_path = Formula["gcc"].opt_bin
-    gcc_version = Formula["gcc"].version.to_s.split(".").first
+    gcc_version = Formula["gcc"].installed_version.major
```
Comment (SeekingMeaning) [inline Formula/pipgrip.rb:56]:
```
irb(main):034:0> gcc.version.to_s.split(".").first
=> "10"
irb(main):035:0> gcc.installed_version.major
=> #<Version::NumericToken 10>
```

File: Formula/dynare.rb [hunk 7]
```diff
@@ -63,7 +63,7 @@ def install
                           "--disable-doc"
     # Octave hardcodes its paths which causes problems on GCC minor version bumps
     gcc = Formula["gcc"]
-    flibs = "-L#{gcc.lib/"gcc"/gcc.version_suffix} -lgfortran -lquadmath -lm"
+    flibs = "-L#{gcc.lib/"gcc"/gcc.installed_version.major} -lgfortran -lquadmath -lm"
```
Comment (SeekingMeaning) [inline Formula/dynare.rb:66]:
```
irb(main):006:0> gcc.version_suffix
=> "10"
irb(main):007:0> gcc.installed_version.major
=> #<Version::NumericToken 10>
```

File: Formula/gcc.rb [hunk 8]
```diff
@@ -34,7 +34,7 @@ def version_suffix
     if build.head?
       "HEAD"
     else
-      version.to_s.slice(/\d+/)
+      version.major.to_s
```
Comment (SeekingMeaning) [inline Formula/gcc.rb:37]:
```
irb(main):009:0> gcc.version.to_s.slice(/\d+/)
=> "10"
irb(main):010:0> gcc.version.major.to_s
=> "10"
```

File: Formula/gcc@4.9.rb [hunk 9]
```diff
@@ -86,7 +86,7 @@ def install
     # Build dependencies in-tree, to avoid having versioned formulas
     resources.each { |r| r.stage(buildpath/r.name) }
 
-    version_suffix = version.to_s.slice(/\d\.\d/)
+    version_suffix = version.major_minor.to_s
```
Comment (SeekingMeaning) [inline Formula/gcc@4.9.rb:89]:
```
irb(main):012:0> Formula["gcc@4.9"].version.to_s.slice(/\d\.\d/)
=> "4.9"
irb(main):013:0> Formula["gcc@4.9"].version.major_minor.to_s
=> "4.9"
irb(main):014:0> Formula["gcc@4.9"].version.major_minor
=> #<Version:0x00007fde991012b8 @version="4.9">
```

File: Formula/gcc@5.rb [hunk 10]
```diff
@@ -75,7 +75,7 @@ def install
     # C, C++, ObjC and Fortran compilers are always built
     languages = %w[c c++ fortran objc obj-c++]
 
-    version_suffix = version.to_s.slice(/\d/)
+    version_suffix = version.major.to_s
```
Comment (SeekingMeaning) [inline Formula/gcc@5.rb:78]:
```
irb(main):016:0> Formula["gcc@5"].version.to_s.slice(/\d/)
=> "5"
irb(main):017:0> Formula["gcc@5"].version.major.to_s
=> "5"
irb(main):018:0> Formula["gcc@5"].version.major
=> #<Version::NumericToken 5>
```

File: Formula/gcc@6.rb [hunk 11]
```diff
@@ -44,7 +44,7 @@ def install
     # C, C++, ObjC, Fortran compilers are always built
     languages = %w[c c++ objc obj-c++ fortran]
 
-    version_suffix = version.to_s.slice(/\d/)
+    version_suffix = version.major.to_s
```
Comment (SeekingMeaning) [inline Formula/gcc@6.rb:47]:
```
irb(main):020:0> Formula["gcc@6"].version.to_s.slice(/\d/)
=> "6"
irb(main):021:0> Formula["gcc@6"].version.major.to_s
=> "6"
irb(main):022:0> Formula["gcc@6"].version.major
=> #<Version::NumericToken 6>
```

File: Formula/gcc@7.rb [hunk 12]
```diff
@@ -31,7 +31,7 @@ def install
     # GCC will suffer build errors if forced to use a particular linker.
     ENV.delete "LD"
 
-    version_suffix = version.to_s.slice(/\d/)
+    version_suffix = version.major.to_s
```
Comment (SeekingMeaning) [inline Formula/gcc@7.rb:34]:
```
irb(main):024:0> Formula["gcc@7"].version.to_s.slice(/\d/)
=> "7"
irb(main):025:0> Formula["gcc@7"].version.major.to_s
=> "7"
irb(main):026:0> Formula["gcc@7"].version.major
=> #<Version::NumericToken 7>
```

File: Formula/gcc@8.rb [hunk 13]
```diff
@@ -31,7 +31,7 @@ def install
     # GCC will suffer build errors if forced to use a particular linker.
     ENV.delete "LD"
 
-    version_suffix = version.to_s.slice(/\d/)
+    version_suffix = version.major.to_s
```
Comment (SeekingMeaning) [inline Formula/gcc@8.rb:34]:
```
irb(main):028:0> Formula["gcc@8"].version.to_s.slice(/\d/)
=> "8"
irb(main):029:0> Formula["gcc@8"].version.major.to_s
=> "8"
irb(main):030:0> Formula["gcc@8"].version.major
=> #<Version::NumericToken 8>
```

File: Formula/gcc@9.rb [hunk 14]
```diff
@@ -32,6 +32,8 @@ def install
     # GCC will suffer build errors if forced to use a particular linker.
     ENV.delete "LD"
 
+    version_suffix = version.major.to_s
```
Comment (SeekingMeaning) [inline Formula/gcc@9.rb:35]:
```
irb(main):032:0> Formula["gcc@9"].version.major.to_s
=> "9"
irb(main):033:0> Formula["gcc@9"].version.major
=> #<Version::NumericToken 9>
```

**Review by SeekingMeaning — COMMENTED** [review]
```
irb(main):001:0> gcc = Formula["gcc"]
```

**Review by SMillerDev — APPROVED** [review]

**Pull request merged**